File: _projects/mmo/server/modules/fight/common/attr.go
package common

// 属性定义
const (
	Level int = iota
	Side
	HP
	HPMax
	Energy
	EnergyMax

	AttackSpeed  // 攻速 基础值受武器影响
	CriticalRate // 暴击概率 [0-1]

	PhysicPower // 物伤
	MagicPower  // 法伤
	PhysicArmor
	MagicArmor

	WeaponMinDmg // 武器最小伤害
	WeaponMaxDmg // 武器最大伤害

	MaxAttrNum // last
)

func init() {
	registerAttrs()
}

func registerAttrs() {
	var R = RegisterAttr
	R(Level, "等级")
	R(Side, "阵营")
	R(HP, "-")
	// 属性定义中一般血量都是hpmax
	R(HPMax, "生命")
	R(Energy, "-")
	R(EnergyMax, "能量")

	R(AttackSpeed, "攻速")
	R(CriticalRate, "暴击")

	R(PhysicPower, "物伤")
	R(MagicPower, "法伤")
	R(PhysicArmor, "护甲")
	R(MagicArmor, "法抗")

	R(WeaponMinDmg, "武器最小伤害")
	R(WeaponMaxDmg, "武器最大伤害")
}

File: _projects/mmo/client/Assets/Scripts/systems/consts/attr.cs

namespace Phoenix.Game
{
    // 属性定义，和服务器对应
    public enum eAttrType
    {
        Invalid = -1,
        Level,
        Side,
        HP,
        HPMax,
        Energy,
        EnergyMax,
        AttackSpeed,
        CriticalRate,
        PhysicPower,
        MagicPower,
        PhysicArmor,
        MagicArmor,
        WeaponMinDmg,
        WeaponMaxDmg,


        MaxAttrNum,
    }

    public static partial class Strings
    {
        public static string[] ATTR_NAMES = new string[]
        {
            "-",
            "-",
            "-",
            "生命",
            "-",
            "能量",
            "攻速",
            "暴击",
            "物伤",
            "法伤",
            "护甲",
            "法抗",
            "武器最小伤害",
            "武器最大伤害",
        };
    }
}